pydra/engine/workers.py

```diff
@@ -112,6 +110,10 @@ def close(self):
         except ValueError:
             # nothing pending!
             pending = set()
+        if done.union(pending) != futures:
```

Comment:
We should probably start converting these checks to assertions

```suggestion
        assert done.union(pending) == futures, "all tasks from futures should be either in done or pending"
```